python/ml4ir/base/pipeline.py

```diff
@@ -137,6 +137,10 @@ def __init__(self, args: Namespace):
         # Set random seeds
         self.set_seeds()
 
+        self.logger.info("Running pre-processing step.")
```

Comment:
@Ullimague Don't you want to also group the steps above and below this call into two separate methods? Does this already help you with your goal?